rmf_fleet_adapter/src/full_control/estimation.cpp

```diff
@@ -260,6 +266,19 @@ void estimate_waypoint(
     TravelInfo& info)
 {
   std::string last_known_map = l.level_name;
+  if(!last_known_map.empty())
+  {
+    const auto starts = rmf_traffic::agv::compute_plan_starts(
+      *info.graph,
+      last_known_map,
+      {l.x, l.y, l.yaw},
+      rmf_traffic_ros2::convert(node->now()));
+
+    assert(starts.size() > 0);
+    info.updater->update_position(starts[0].waypoint(), l.yaw);
```

Comment:
In the absence of any information besides the location information, it's best to use [this `update_position` signature](https://github.com/osrf/rmf_core/blob/4180a944e7497a3dd6b08f17280706ce6aca8a30/rmf_fleet_adapter/include/rmf_fleet_adapter/agv/RobotUpdateHandle.hpp#L79) because it will be effectively the same as what you're doing here, except it will take all possible starts into account.

That being said, this also throws away information about the last known map and last known waypoint. If this is an improvement on the previous implementation, then we should probably be completely throwing out the old implementation.